common/djangoapps/student/models.py

```diff
@@ -1589,3 +1589,11 @@ def user_can_skip_entrance_exam(cls, user, course_key):
             except EntranceExamConfiguration.DoesNotExist:
                 can_skip = False
         return can_skip
+
+
+class LanguageProficiency(models.Model):
+    """
+    Represents a user's language proficiency.
+    """
+    user_profile = models.ForeignKey(UserProfile, db_index=True, related_name='language_proficiencies')
+    code = models.CharField(max_length=16, db_index=True, choices=settings.ALL_LANGUAGES, blank=False)
```

Comment:
I wonder if we should put some big disclaimer in the settings file that ALL_LANGUAGES should never be changed because it will impact existing database models.